nengo/spa/thalamus.py

```diff
@@ -188,6 +189,8 @@ def add_route_effect(self,
             subdim = self.subdim_channel
             if isinstance(target_module, nengo.spa.Buffer):
                 subdim = target_module.state.dimensions_per_ensemble
+            elif isinstance(source_module, nengo.spa.Buffer):
+                subdim = source_module.state.dimensions_per_ensemble
```

Comment:
What do you want it to do in those cases?  What it does right now is use the Buffer's geometry for storing information if it can find it, and otherwise it falls back on the parameter its been told to use.

I could see just getting rid of that check and always using the subdim_channel parameter.....  but that'd be a performance change from this PR, and my goal here was just to fix a bug that's breaking Spaun.  :)